components/tidb_query/src/expr/ctx.rs

```diff
@@ -71,6 +71,8 @@ pub struct EvalConfig {
     // warning is a executor stuff instead of a evaluation stuff.
     pub max_warning_cnt: usize,
     pub sql_mode: SqlMode,
+
+    pub encode_type: EncodeType,
```

Comment:
I don't think `encode_type` is needed in the eval context. I mean, there should be no UDFs or executors need to know the format to encode.

Reply:
I move the field `encode_type` to the struct `BatchExecutorsRunner` . It it ok?

Reply:
Yes, I think it is the correct direction.